content/en/blog/wireshark.md

```diff
@@ -119,63 +123,90 @@ service PersonSearchService {
-Person objects can be return back to client one after another:
-![wireshark_grpc_protobuf_search_response](/img/wireshark_grpc_protobuf_search_response.png)
-
-You may have noticed that the `portrait_image` of bytes type is parsed as a
-PNG data. It's because we register the PNG dissector in the `"protobuf_field"`
-dissector table for parsing the value of field `portrait_image` as picture by
-putting following Lua script `'protobuf_portrait_field.lua'` into the 'plugins'
-subdirectory of 'Personal configuration' directory:
-
-```lua
-do
-    local protobuf_field_table = DissectorTable.get("protobuf_field")
-    local png_dissector = Dissector.get("png")
-    protobuf_field_table:add("tutorial.Person.portrait_image", png_dissector)
-end
-```
+Because the service uses the `Person` type defined in `addressbook.proto`,
+the address book `.proto` is imported at the start of the file.
+
+### Setting protobuf search paths
+
+Wireshark gives the most meaningful decodings when it knows about the `.proto`
+files used by the apps whose messages you are analyzing.
+
+You can tell Wireshark where to find `.proto` files by setting the **Protobuf
+Search Paths** in the preferences accessible from the **Edit** menu under
+**Preferences \> Protocols \> Protobuf**.
+
+If our example app's `.proto` files are in the `d:/protos/my_proto_files` directory,
+and the official Protobuf library directory is
+`d:/protos/protobuf-3.4.1/include`, then add these two paths as _source
+directories_ like this:
+
+![Protobuf-search-paths dialog](/img/wireshark_protobuf_search_paths.png)
+
+By selecting the **Load all files** option for the app's protocol directory you
+enable preloading of message definitions from the `addressbook.proto` file.
+
+### Loading a capture file
+
+From the Wireshark [SampleCaptures page][], download the following sample gRPC
+capture file created by running the app and issuing a search request:
+[grpc_person_search_protobuf_with_image.pcapng][].
+
+Load this file in Wireshark by selecting **Open** from the **File** menu.
+
+### Setting port traffic type
+
+The app's gRPC traffic was captured on port 50051. Before you can decode the
+port's messages, you need to register the port as carrying HTTP2 traffic; do
+this through the **Decode As** dialog, which you access from the **Analyze**
+menu:
+
+![Decode-as dialog](/img/wireshark_decode_as_dialog.png)
+
+### Decoding the search request message
+
+<!--
+TODO: What are the steps that the user needs to follow before the user can see
+      the decoded message as shown in the screenshot?
+-->
+Select the first gRPC message sent to port 50051, it corresponds to the sample's
+service request message. This is how Wireshark dissects the gRPC request:
+
+<!--
+TODO: I suggest reverting to the layout of the previous image, which didn't
+      show the bytes pane. Also collapsing the "Stream: DATA" would be a good idea.
+-->
+![Decoded search request](/img/wireshark_grpc_protobuf_search_request.png)
```

Comment:
OK. I will update new request and response screenshots.

Reply:
Hi, @chalin I've captured several request and response images. Could you choose the images you think are more suitable, and then upload them to git directly?

>Note that first request and response are filtering with 'http2' and others with 'grpc'.

Request screenshots:
![wireshark_grpc_protobuf_search_request1](https://user-images.githubusercontent.com/30267326/105361690-e847b380-5c34-11eb-884e-b835be9e3f70.png)

![wireshark_grpc_protobuf_search_request2](https://user-images.githubusercontent.com/30267326/105361703-ebdb3a80-5c34-11eb-8258-8aaee8134a86.png)

![wireshark_grpc_protobuf_search_request3](https://user-images.githubusercontent.com/30267326/105361705-ec73d100-5c34-11eb-9cfd-a49e5b04d49d.png)

Response screenshots:
![wireshark_grpc_protobuf_search_response1](https://user-images.githubusercontent.com/30267326/105361711-eda4fe00-5c34-11eb-8956-f41284f63ccb.png)

![wireshark_grpc_protobuf_search_response2](https://user-images.githubusercontent.com/30267326/105361720-eed62b00-5c34-11eb-89cc-85fac12f7796.png)

![wireshark_grpc_protobuf_search_response3](https://user-images.githubusercontent.com/30267326/105361722-ef6ec180-5c34-11eb-9c51-b2994b355fe5.png)

Reply:
> you choose the images you think are more suitable, and then upload them to git directly?

Yes, thanks for the images, I'll choose from those.

Reply:
Could you give me a screenshot of Wireshark **before** setting "Decode As", in the layout of the first image from the image list above? That will allow us to explain the screen layout / panes. Thanks.

Reply:
More images for choosing: ^_^

![wireshark_before_decode_as](https://user-images.githubusercontent.com/30267326/105365185-f0a1ed80-5c38-11eb-9175-186c07386b99.png)

![wireshark_before_decode_as2](https://user-images.githubusercontent.com/30267326/105364925-ade01580-5c38-11eb-9c02-de696999557b.png)

![wireshark_before_decode_as3](https://user-images.githubusercontent.com/30267326/105364946-b3d5f680-5c38-11eb-8007-2425d3059616.png)

![wireshark_decode_as_above_mainwin](https://user-images.githubusercontent.com/30267326/105364952-b6d0e700-5c38-11eb-8fcb-8c30e219dfc1.png)

![wireshark_search_paths_on_mainwin](https://user-images.githubusercontent.com/30267326/105364957-ba646e00-5c38-11eb-9328-bf5ee85d4f28.png)